project/src/web/app/src/components/ui/button/FriendRequestButton.vue

```diff
@@ -118,43 +174,40 @@ export default defineComponent({
           }
         )
         .then((response) => {
-          if (response.data.friendship.accepted == true)
-            this.isFriend = Friends.TRUE;
-          else if (response.data.friendship.accepted == false)
-            this.isFriend = Friends.PENDING;
-          else this.isFriend = Friends.FALSE;
+          this.isFriend = FriendsRequestStatus.PENDING;
         })
         .catch((error) => {
           if (error.response?.status == 400) {
             console.log(error.response?.data);
-            this.isFriend = Friends.TRUE;
+            this.isFriend = FriendsRequestStatus.TRUE;
           }
         });
     },
-    remove() {
+    async remove() {
       // Remove the current user as friend of the user we are on if friend
-      const response = axios
+      const response = await axios
         .delete(
           `${import.meta.env.VITE_APP_API_URL}/friends/${
-            this.friendId
+            this.friendshipId
```

Comment:
Here I believe it should be `this.receiver` as `this.friendshipId` is not an Id of a user and the `/unfriend` endpoint expect a userId. I've changed it locally and it was working

Reply:
make the change for unfriend 

```js
.delete(
          `${import.meta.env.VITE_APP_API_URL}/friends/${
            this.friendId
          }/unfriend`,
          {
            withCredentials: true,
          }
        )
```